src/carts/src/carts-service/dynamo_setup.py

```diff
@@ -0,0 +1,61 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+import os
+import time
+import boto3
+import requests
+import json
+from server import app
+
+
+
+
+# DynamoDB table names passed via environment
+ddb_table_carts = os.getenv("DDB_TABLE_CARTS")
+
+# Allow DDB endpoint to be overridden to support amazon/dynamodb-local
+ddb_endpoint_override = os.getenv("DDB_ENDPOINT_OVERRIDE")
+running_local = False
+
+dynamo_client = None
+
+def verify_local_ddb_running(endpoint):
+    app.logger.info(f"Verifying that local DynamoDB is running at: {endpoint}")
+    for _ in range(5):
+        try:
+            response = requests.get(endpoint)
+            if response.status_code >= 200:
```

Comment:
This should be more specific so as to only consider response codes within the 200's as any 400 or 500 errors would not be deemed a success